drivers/platform/maxim/irq_extra.h

```diff
@@ -39,21 +39,9 @@
 #ifndef IRQ_MAXIM_EXTRA_H
 #define IRQ_MAXIM_EXTRA_H
 
-#include "max32660.h"
```

Comment:
are these two changes related? (including max_hal and removing irq_id)? if not, separate commits should be added.

Reply:
They are not related, I split the commits.